### PR TITLE
SQL-504: Use tag-triggered releases on evergreen

### PR DESCRIPTION
**Pull request by rychipman**
The nature of this change is such that I won't _really_ be able to know it works until I actually commit it, but all the manual testing I've been able to do with patches and local runs of gradle seem to suggest that this will work. My plan is to cut a `2.0.0-beta-1` release as soon as this is merged to see if it works.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -36,27 +36,36 @@ Close the release on JIRA, adding the current date (you may need to ask the SQL
...
+git push --tags
+```
+This should trigger an Evergreen version that can be viewed on the [mongo-jdbc-driver waterfall](https://evergreen.mongodb.com/waterfall/mongo-jdbc-driver).
+If it does not, you may have to ask the project manager to give you the right permissions to do so.
+Make sure to run the 'release' task, if it is not run automatically.
+
+#### Set Evergreen Priorities
+Some evergreen variants may have a long schedule queue.
+To speed up release tasks, you can set the task priority for any variant to 101 for release candidates and 200 for actual releases.
+If you do not have permissions to set priority above 100, ask someone the project manager to set the
```
Comment (bucaojit) [inline RELEASE.md:52]:
```suggestion
If you do not have permissions to set priority above 100, ask the project manager to set the
```

**Review by bucaojit — APPROVED** [review]
Looks good.

**Review by nbagnard — COMMENTED** [review]
LGTM :)
Only a few comments regarding extra clean-up which could be integrated before pushing the changes.

File: build.gradle [hunk 2]
```diff
@@ -8,7 +8,6 @@ plugins {
 
 ext {
     gitVersion = getGitVersion()
```
[the remark this answers is not in the document]
Comment (rychipman) [reply build.gradle:10]: good catch! deleted

File: build.gradle [hunk 3]
```diff
@@ -157,24 +156,64 @@ dependencies {
...
+		dependsOn ':publishToSonatype'
+		// Make sure we publish to staging first
+		project.tasks.getByName('publishToSonatype').finalizedBy(
+				project.tasks.getByName('closeAndReleaseRepository'),
+		)
+	}
+}
+
+def getReleaseVersion() {
+	def version = getGitVersion()
```
[the remark this answers is not in the document]
Comment (rychipman) [reply build.gradle:180]: you're right -- not needed anymore!

File: build.gradle [hunk 4]
```diff
@@ -157,24 +156,64 @@ dependencies {
...
+def getGitVersion() {
+    def out = new ByteArrayOutputStream()
     exec {
-        commandLine 'git', 'log', '-1', '--pretty=%B'
-        standardOutput = describeStdOut
+        commandLine 'git', 'describe', '--tags', '--always', '--dirty'
+        standardOutput = out
     }
-    describeStdOut.toString().trim()
+    def version = out.toString().substring(1).trim()
```
[the remark this answers is not in the document]
Comment (rychipman) [reply build.gradle:214]: This is what we actually use for getting the function's return value -- we take the stdout of the execed `git` command, and turn it into a string!

**Comment by rychipman** [comment]
evergreen merge

File: build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (nbagnard) [reply build.gradle:212]: @rychipman : What I meant is that "version" is a local variable here.

**Pull request merged**